aws-cfn/cross-service/resilient-workflow/README.md

```diff
@@ -0,0 +1,240 @@
+2. An instance profile and an associated role and policy that grants permission to
+   instances to access DynamoDB and Systems Manager.
+3. A launch template that specifies the instance profile and a startup script
+   that starts a Python web server on each instance.
+4. An Auto Scaling group that starts EC2 instances, one in each of three
+   Availability Zones.
+5. An Application Load Balancer that handles HTTP requests to a single endpoint.
+6. A target group that connects the load balancer to instances in the Auto Scaling group.
+7. A listener that is added to the load balancer and forwards requests to the target group.
+
+To deploy these resources, you will need to provide several pieces of information about your AWS Account.
+
+| Parameter | AWS CLI                                                                                                      |
+| --------- | ------------------------------------------------------------------------------------------------------------ |
+| `VPCId`   | `aws ec2 describe-vpcs --filters Name=is-default,Values=true --query "Vpcs[0].VpcId"`                        |
+| `SGId`    | `aws ec2 describe-security-groups --group-names default --query "SecurityGroups[0].GroupId"`                 |
+| `Subnets` | `aws ec2 describe-subnets --filters Name=vpc-id,Values="vpc-0abcdef1234567890" --query "Subnets[].SubnetId"` |
+
+After running these commands, replace the placeholders in `params.json` with the returned values.
+
+```
+aws cloudformation create-stack \
+  --stack-name resilient-service \
+  --template-body file://./resilient-service.yaml \
+  --parameters: file://./params.json \
+  --capabilities CAPABILITY_NAMED_IAM
+```
+
+After creating the stack, you can find the URL for the Load Balancer in the stack's parameters.
+
+```
+aws cloudformation describe-stacks --stack-name resilience-demo --query 'Stacks[0].Outputs[?OutputKey==`LB`].OutputValue | [0]'
+
+# "doc-example-resilience-lb-0123456789.us-east-1.elb.amazonaws.com"
+```
+
+You can then access the URL via your browser or using `curl`.
+
+##### Demonstrate resiliency
+
+This part of the example demonstrates resiliency by simulating several kinds of failures.
+It uses Systems Manager parameters to update how the web server responds to requests. It
+also uses health checks to show how to make your web server more resilient to failure.
+
+Along with recommendations returned by the DynamoDB table, the web service includes the
+instance ID and Availability Zone so you can see how the load balancer distributes
+requests among the instances in the Auto Scaling group.
+
+The scenario takes the following steps. After editing the `params.json` file for each step, update the stack
+to see the changes by running an `update-stack` command.
+
+```
+aws cloudformation update-stack \
+  --stack-name resilient-service \
+  --template-body file://./resilient-service.yaml \
+  --parameters: file://./params.json \
+  --capabilities CAPABILITY_NAMED_IAM
+```
+
```

Comment:
It might also be interesting to include instructions on how to get the health of the target instances, which you can do with cli `aws elbv2 describe-target-health --target-group-arn <tg-arn>`

Reply:
Done.